contracts/ArrakisV2.sol

```diff
@@ -472,24 +472,22 @@ contract ArrakisV2 is IUniswapV3MintCallback, ArrakisV2Storage {
         emit LogRebalance(rebalanceParams_);
     }
 
-    function _withdraw(
-        IUniswapV3Pool pool_,
-        int24 lowerTick_,
-        int24 upperTick_,
-        uint128 liquidity_
-    ) internal returns (Withdraw memory withdraw) {
+    function _withdraw(IUniswapV3Pool pool_, BurnLiquidity memory burn_)
+        internal
+        returns (Withdraw memory withdraw)
+    {
         (withdraw.burn0, withdraw.burn1) = pool_.burn(
-            lowerTick_,
-            upperTick_,
-            liquidity_
+            burn_.range.lowerTick,
+            burn_.range.upperTick,
+            burn_.liquidity
         );
 
         (uint256 collect0, uint256 collect1) = pool_.collect(
             address(this),
-            lowerTick_,
-            upperTick_,
-            type(uint128).max,
-            type(uint128).max
+            burn_.range.lowerTick,
+            burn_.range.upperTick,
+            burn_.amount0Max,
```

Comment:
If the goal is to have finer slippage control for each pool, it still should be `burn_.amoun0Min` and `burn_.amount1Min`. The check should be done after L483 and before L485:

```solidity
require(withdraw.burn0 >= burn_.amoun0Min, "B0");
require(withdraw.burn1 >= burn_.amoun1Min, "B1");
```